Tutorial Fixes

koalateectrl opened the pull request:
Was going through the Miden tutorials to learn how to use it and thought I would provide some minor fixes while at it.

One small recurring thing was that the folder structure causes the code snippets to require a path of "./..." for the MASM files while the full examples require "../...".

There's also a minor aethestic issue where the comments are causing the code snippet to render far down in the box (lots of space before the code): see [code for Step 3](https://0xmiden.github.io/miden-docs/imported/miden-tutorials/src/rust-client/mint_consume_create_tutorial.html), but I didn't want to touch that since I wasn't sure how you wanted to fix that.

mmagician commented:
@koalateectrl could you rebase please? I think some of the changes from this PR have already been addressed in https://github.com/0xMiden/miden-tutorials/pull/121, but would be good to include the remaining fixes you proposed here

partylikeits1983 commented:
@koalateectrl Looking great! ❤️❤️

Could you resolve the conflict and add this as the counter contract in the markdown files & `masm/accounts/counter.masm`?  

```
use.miden::account
use.std::sys

const.COUNTER_SLOT=0

#! Inputs:  []
#! Outputs: [count]
export.get_count
    push.COUNTER_SLOT
    # => [index]

    exec.account::get_item
    # => [count]

    # clean up stack
    movdn.4 dropw
    # => [count]
end

#! Inputs:  []
#! Outputs: []
export.increment_count
    push.COUNTER_SLOT
    # => [index]

    exec.account::get_item
    # => [count]

    add.1
    # => [count+1]

    debug.stack

    push.COUNTER_SLOT
    # [index, count+1]

    exec.account::set_item
    # => [OLD_VALUE]

    dropw
    # => []
end
```

koalateectrl commented:
@mmagician @partylikeits1983 I just rebased, fixed the merge conflict, and updated the counter.masm file. Note that since we have a new counter.masm file, you may want to also update the corresponding files such as docs/src/rust-client/counter_contract_tutorial.md to explain this new counter.masm file. Not sure if you want to do that in this PR or another one.

mmagician commented on docs/src/rust-client/network_transactions_tutorial.md:
```diff
@@ -928,9 +925,15 @@ Add this code to your `main()` function:
 #     // STEP 3: Deploy Network Account with Transaction Script
 #     // -------------------------------------------------------------------------
 #     println!("\n[STEP 3] Deploy network counter smart contract");
+<<<<<<< HEAD
```
merge leftover

mmagician commented on docs/src/rust-client/public_account_interaction_tutorial.md:
```diff
@@ -70,39 +70,38 @@ Inside of the `masm/accounts/` directory, create the `counter.masm` file:
-    # => [count+1]
-    push.0
-
+    push.COUNTER_SLOT
```
I would remove `debug.stack` from our tutorials across the repo @partylikeits1983

mmagician commented on docs/src/rust-client/public_account_interaction_tutorial.md:
```diff
@@ -70,39 +70,38 @@ Inside of the `masm/accounts/` directory, create the `counter.masm` file:
     # => []
+
     exec.sys::truncate_stack
+    # => []
```
I think we decided to forego `truncate_stack`, please check

mmagician commented on masm/accounts/counter.masm:
```diff
@@ -32,8 +35,8 @@ export.increment_count
+    dropw
     # => []
-end
+end
```
newline missing

mmagician replied:
Ping

mmagician commented on masm/accounts/counter.masm:
```diff
@@ -3,19 +3,22 @@ use.std::sys
-    exec.sys::truncate_stack
-    # => []
+    # clean up stack
+    movdn.4 dropw
```
Is this needed?

mmagician replied:
my understanding when we [decided to remove `truncate_stack`](https://github.com/0xMiden/miden-tutorials/pull/117#discussion_r2341828044) was that the stack is empty at the end. Why do we suddenly have padding?

mmagician reviewed (commented):
Two small leftover comments, otherwise LGTM

partylikeits1983 commented:
> @partylikeits1983 I just pasted your counter.masm file into masm/accounts/counter.masm, but like I mentioned, for consistency sake, you would want to go back and update all the .md files in the docs/src/rust-client files to reflect your new MASM file like the ones @mmagician pointed out above, and especially the docs/src/rust-client/counter_contract_tutorial.md file, where you are explaining what each line of code in the MASM file actually is doing.

I think its pretty easy to update the steps to explain what is going on. Could you update `rust-client/counter_contract_tutorial.md`, `rust-client/oracle_tutorial.md`, and `web-client/counter_contract_tutorial.md`?

Updated steps:



```
#### Here's a breakdown of what the `get_count` procedure does:

1. Pushes `0` (COUNTER_SLOT) onto the stack, representing the index of the storage slot to read.
2. Calls `account::get_item` with the index of `0`.
3. Uses `movdn.4` to move `count` down to be the 5th element on the stack, then using `dropw` drops four `0`s from the top of the stack to ensure a stack depth of 16 on return from the procedure.
4. The value returned from `account::get_item` is on the top of the stack and will be returned when this procedure is called.

#### Here's a breakdown of what the `increment_count` procedure does:

1. Pushes `0` (COUNTER_SLOT) onto the stack, representing the index of the storage slot to read.
2. Calls `account::get_item` with the index of `0`.
3. Pushes `1` onto the stack.
4. Adds `1` to the count value returned from `account::get_item`.
5. _For demonstration purposes_, calls `debug.stack` to see the state of the stack
6. Pushes `0` (COUNTER_SLOT) onto the stack, which is the index of the storage slot we want to write to.
7. Calls `account::set_item` which saves the incremented count to storage at index `0`
8. Uses `dropw` to drop the first four values from the top of the stack to ensure a stack depth of 16 on return from the procedure.
```

partylikeits1983 commented:
@koalateectrl 

To format the markdown you can run `npx prettier --write "**/**.md"` in the docs directory. This will make the CI check pass.

To make the docs test pass, you will need to update the counter contract deployment:

1) in the `rust-client` directory, run `cargo run --release --bin counter_contract_deploy`
2) update the old counter contract address across the docs, with the new counter contract address printed out from `counter_contract_deploy` program.

partylikeits1983 reviewed (approved):
Approving this PR. Before merging, lets get the markdown formatting check & doc tests to pass, otherwise looks good. Once the CI is passing, @mmagician could you do one final check on this PR?

mmagician commented on docs/src/rust-client/counter_contract_tutorial.md:
```diff
@@ -728,15 +723,17 @@ async fn main() -> Result<(), ClientError> {
 
 [STEP 2] Call Counter Contract With Script
-Stack state before step 2502:
+one or more warnings were emitted
```
```suggestion
```

mmagician commented on masm/accounts/counter.masm:
```diff
@@ -32,8 +35,8 @@ export.increment_count
+    dropw
     # => []
-end
+end
```
```suggestion
end

```
